fix(api): work-preview place data

koeaw opened the pull request:
Change place_of_publication back to ListField in
ExpressionDataSerializer (used by WorkPreviewSerializer, gets overridden by ExpressionDataDetailSerializer).

koeaw committed:
fix(api): work-preview place data

Change place_of_publication back to ListField in
ExpressionDataSerializer (used by WorkPreviewSerializer,
gets overridden by ExpressionDataDetailSerializer).

koeaw commented:
@sennierer I think this should fix places for the `work-preview` endpoint.

I.e. `PlaceDataSerializerMin` and `PlaceDataSerializer` (_both of which_ are used by the detail endpoint, which I hadn't realised) don't need adjusting after all; the preview endpoint just needs to not use them anymore.

This changes e.g. `api/work-preview/26488/` from: 

```json
            "place_of_publication": [
                {
                    "relation_type": null,
                    "latitude": null,
                    "longitude": null
                },
                {
                    "relation_type": null,
                    "latitude": null,
                    "longitude": null
                }
            ]
```

back to a simple list of strings:

```json
            "place_of_publication": [
                "Klagenfurt",
                "Wien"
            ]
```

sennierer reviewed (approved):
This works as intended. However, we need to talk to the frontend and see if they need at least IDs of the places also in the preview? I guess they will at least want to have links on the places they show in the preview view.

koeaw commented:
> This works as intended. However, we need to talk to the frontend and see if they need at least IDs of the places also in the preview? I guess they will at least want to have links on the places they show in the preview view.

I saw the serializer which does this for publishers in the detail endpoint and moved it up top and renamed it to `NameAndIdSerializer` for clarity, figuring we may wanna reuse it for other stuff.

I'll merge this now as it fixes the current unhelpful output.
